src/Conways/Game/Of/Life/Grid.java

```diff
@@ -3,34 +3,68 @@
 public class Grid {
 
     /**
-     *
      * Instantiates the grid and decides if the cell will be alive
      * or dead in the next generation
-     *
-     * Author: Chashie Komendant
+     * <p>
+     * Author: Jennifer Komendant
      */
 
     private final int row = 30;
     private final int col = 50;
     private final boolean[][] gridArray = new boolean[row][col];
+    private int generation = 0;
 
     public Grid() {
 
     }
 
-    public void populateGrid(){
-
+    public void goToNextGeneration() {
+        for (int i = 0; i < row - 1; i++) {
+            for (int j = 0; j < col - 1; j++) {
+                int aliveNeighbors = 0;
+                for (int k = -1; k <= 1; k++) {
+                    for (int l = -1; l <= 1; l++) {
+                        if (gridArray[i + k][j + l]) {
+                            aliveNeighbors++;
+                        }
+                    }
+                }
+                if (gridArray[i][j] && aliveNeighbors < 2) {
+                    gridArray[i][j] = false;
+                } else if (gridArray[i][j] && aliveNeighbors > 3) {
+                    gridArray[i][j] = false;
+                } else if (gridArray[i][j] && (aliveNeighbors == 2 || aliveNeighbors == 3)) {
+                    gridArray[i][j] = true;
+                } else {
+                    gridArray[i][j] = gridArray[i][j];
+                }
+            }
+        }
+        generation++;
     }
 
-    public void goToNextGeneration(){
-
+    public void clearGrid() {
+        for (int i = 0; i < row; i++) {
+            for (int j = 0; j < col; j++) {
+                gridArray[row][col] = false;
+            }
+        }
+        generation = 0;
     }
 
-    public boolean getCells(int row, int col){
+    public boolean getCells(int row, int col) {
         return gridArray[row][col];
     }
 
-    public void setCells(int row, int col, boolean value){
+    public void setCells(int row, int col, boolean value) {
         gridArray[row][col] = value;
     }
+
+    public int getGeneration() {
+        return generation;
+    }
```

Comment:
What's this for?

Reply:
This is if we decide to make a little box that shows up which generation number we're up to